### ISSUE TITLE
Tidal installer for Mac

### ISSUE THREAD

**Issue by yaxu** [tidalcycles/Tidal#282]
It'd be great if there was an easy way of installing for mac. (see #283 for the equivalent issue for windows and #284 for linux)

There is this script made by the excellent @davidgranstrom:
  https://davidgranstrom.github.io/tidal-bootstrap/ https://github.com/davidgranstrom/tidal-bootstrap
along with this video:
  https://www.youtube.com/watch?v=dJTfGv2sT-w

I'm not 100% sure it works at the moment (I don't have a mac), and it isn't linked from the [install instructions](https://tidalcycles.org/getting_started.html#macos). It'd be good to bring it into the tidalcycles github organisation for long term maintenance. 

The next level would be a one-click download that included everything you need (supercollider, superdirt + dependences, ghc, tidal + dependencies) and installed it all for you.

Any mac-based tidalcyclists up for taking this on?

**Comment by lol-russo** [comment]
Just confirming that @davidgranstrom's install script (and CLI instructions) worked for me on macOS High Sierra v10.13.3 on a late 2015 iMac. Ran it just then and didn't have any issues 👍 

Agree that a one-click install would be amazing, sorry I don't have time to take this on right now though

**Comment by yaxu** [comment]
That's great news @lol-russo, we should definitely link to it from the website, then!

**Comment by yaxu** [comment]
@vism2889 Did you have to install a particular haskell version first or did it work from scratch? I guess we could close this issue, but still think a drag-and-drop installer would be nice, longer term.

**Comment by vism2889** [comment]
The only issue i had was with python versioning, I changed the head of the file to : `#!/usr/bin/env python2` to force python2, and it worked.  I'm using ghci 8.0.2. Don't know much about drag-and-drop / dmg packaging, also think that would be nice.

**Comment by songproducer** [comment]
I had some issues installing which I was able to fix after searching around. More details here: https://steemit.com/music/@djleo/programming-music-how-i-got-tidal-working-on-mac

**Comment by vism2889** [comment]
@yaxu https://github.com/tidalcycles/tidal-bootstrap/issues/10 Would like to propose this change to bootstrap code so it defaults to python2, which is the only compatible python for this script at the moment, and from the best of my knowledge, and searching, is the version of python that is factory installed on OSX.

**Comment by yaxu** [comment]
I've accepted your PR @vism2889, thanks. I hear that installing the full version of haskell platform 8.4.1 works without the 'no-pie' error. If anyone has time to look at fixing up the script that'd be great!

**Comment by vism2889** [comment]
@yaxu  I have been unable to find a way to specify Haskell versions (8.4.1) with `brew cask install`, for examplei dont know if: `brew cask install haskell-platform-8.4.1` will work.  The bootstap script now has `ghci` as a target variable, I think that could be changed to `haskell-platform`.  Im not sure what that defaults to though, assuming the newest stable release, which may or may not be 8.4.1 at this point.

**Comment by dandrewstewart** [comment]
I will watch this thread for any news of compatibility with latest Haskell Platform (version 8.4.3).  I went with the "manual" installation instructions, not bootstrap option.  After a failed attempt, and finding issue, "**Windows installation problem. Failed to connect to tidal server. Try specifying a different port**", which is now closed, I can confirm that version 8.2.1 is required.  Thank you Kazuki-Tohyama !
